lib/cabal-project-parser.nix

```diff
@@ -171,32 +171,20 @@ let
           mkdir -p $out
           hackage-to-nix $out ${home}/.cabal/packages/${name}/01-index.tar ${attrs.url}
         '');
-      # Tarball info in the same format as `extra-hackage-tarballs` passed in to cabalProject
-      tarball = {
-        ${name} = home + "/.cabal/packages/${name}/01-index.tar.gz";
+      # Directory to `lndir` when constructing a suitable $HOME/.cabal dir
+      repo = {
+        ${name} = home + "/.cabal/packages/${name}";
```

Comment:
This is worth fixing because currently the `sha256` for repository includes the `.cabal/packages/${name}` directory name.  Two projects pointing to the same repository, but using different names would have require different `sha256`.  Unfortunately this will invalidate current `sha256` values, but better to fix it now.